addons/dynamishelper/dynamishelper.lua

```diff
@@ -24,408 +24,427 @@
+function init_window()
+    local showCurrenciesDivider = false
+    window.text(window,Yellow.."Time remaining: ${time|initializing...}")
+    window.appendline(window,"\\cr————————————————————")
+    window.appendline(window,"${current_mob|(unknown)}\n"..Green.."${current_proc|(none)}")
+    window.appendline(window,"\\cr————————————————————")
+    for currency in Currencies:it() do
+        if w[currency] > 0 then
+            showCurrenciesDivider = true
+            if currency == "Ordelle Bronzepiece" or currency == "Montiont Silverpiece" then
+                window.appendline(window, currency..": ${"..currency.."|0}")
+            elseif currency == "One Byne Bill" or currency == "One Hundred Byne Bill" then
+                window.appendline(window, currency..": ${"..currency.."|0}")
+            elseif currency == "Tukuku Whiteshell" or currency == "Lungo-Nango Jadeshell" then
+                window.appendline(window, currency..": ${"..currency.."|0}")
+            else
+                window.appendline(window, "\\cr"..currency..": ${"..currency.."|0}")
+            end
+        end
+    end
+    if showCurrenciesDivider then
+        window.appendline(window,"\\cr————————————————————")
+    end
+    for granule in Granules:it() do
+        if(w[granule] == 1) then
+            window.appendline(window, Green..granule)
+        else
+            window.appendline(window, Red..granule)
+        end
+    end
+end
+init_window()
+
+-------------------------------------------------------------------------------
+-- Register a prerendere event for the display refresh ------------------------
+-------------------------------------------------------------------------------
+windower.register_event('prerender', function()
+	if obj_time < 1 or obj_time == end_time - os.time() then
+        return
+    end
+	
+	obj_time = end_time - os.time()
+	w.time = os.date('!%H:%M:%S', obj_time)
+	window:update(w)
+end)
+-------------------------------------------------------------------------------
+-- Did we enter a Dynamis Zone? -----------------------------------------------
+-------------------------------------------------------------------------------
+windower.register_event('zone change', function(zone)
+    if ProcZones:contains(zone) then
+        init_currency()
+        init_granules()
+        window:show()
+    else
+        window:hide()
+    end
+end)
 
-settings = config.load()
-timer = settings['timer']
-tracker = settings['tracker']
-trposx = settings['trposx']
-trposy = settings['trposy']
-proc = settings['proc']
-pposx = settings['pposx']
-pposy = settings['pposy']
+window:hide()
 
-for i=1, #Currency do
-     Currency[Currency[i]] = 0
+-------------------------------------------------------------------------------
+-- Check if we are in Dynamis and show the overlay ----------------------------
+-------------------------------------------------------------------------------
+if ProcZones:contains(windower.ffxi.get_info().zone) then
```

Comment:
I feel like this `if` block can be removed and the `zone change` and `load` and `login` events should be merged. Would save on a bit of code duplication and also put the initialization logic all in one place, since they are all used to determine initialization.